.github/workflows/kotlin.yml

```diff
@@ -18,19 +18,31 @@ concurrency:
 jobs:
   # See rust.yml how we build, package and release connlib as an example
   kotlin_build:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.runs-on }}
+    strategy:
+      matrix:
+        runs-on:
+          - ubuntu-latest
     defaults:
       run:
         working-directory: ./kotlin/android
     steps:
       - uses: actions/checkout@v4
+      - name: Update toolchain
+        run: rustup show
+      - uses: Swatinem/rust-cache@v2
+        with:
+          workspaces: ./rust
+          shared-key: ${{ runner.os }}
+          save-if: ${{ github.ref == 'refs/heads/main' }}
       - uses: actions/setup-java@v3
         with:
           distribution: temurin
           java-version: 11
       - name: Validate Gradle wrapper
         uses: gradle/wrapper-validation-action@v1
-      - uses: actions/cache@v3
+      - uses: actions/cache/restore@v3
```

Comment:
Isn't `actions/cache@v3` will do the same job except that you won't need to do `actions/cache/save@v3` yourself?

Reply:
This is so that we don't pollute the cache from PRs. I had to split it up to only write the cache on main